File: go/difference-of-squares/difference_of_squares.go
package diffsquares

// SquareOfSums calculates the square of sum of 1 to n
func SquareOfSums(n int) int {
	return (n * n * (n + 1) * (n + 1)) / 4
}

// SumOfSquares calculates the sum of the squares of 1 to n
func SumOfSquares(n int) int {
	return (n * (n + 1) * (2*n + 1)) / 6
}

// Difference calculates the difference between SquareOfSums and SumOfSquares
func Difference(n int) int {
	return SquareOfSums(n) - SumOfSquares(n)
}

File: typescript/difference-of-squares/difference-of-squares.ts
class Squares {
  private readonly num: number

  constructor(num: number) { this.num = num }

  public get squareOfSum(): number {
    return (this.num ** 4 + 2 * this.num ** 3 + this.num ** 2) / 4
  }

  public get sumOfSquares(): number {
    return (this.num + 3 * this.num ** 2 + 2 * this.num ** 3) / 6
  }

  public get difference(): number {
    return this.squareOfSum - this.sumOfSquares
  }
}

export default Squares
